lib/iris/fileformats/name_loaders.py

```diff
@@ -193,19 +193,22 @@ def _calc_integration_period(time_avgs):
     """
     integration_periods = []
     pattern = re.compile(
-        r'(\d{0,2})(day)?\s*(\d{1,2})(hr)?\s*(\d{1,2})(min)?\s*(\w*)')
+        r'\s*(\d{1,2}day)?\s*(\d{1,2}hr)?\s*(\d{1,2}min)?\s*(\w*)\s*')
```

Comment:
The regex allows for any number of trailing whitespace characters at the end of the time string, but this behaviour is not tested. You could add a test that just confirms that the pattern is still appropriately matched with trailing whitespace.